geomag/world_magnetic_model.py

```diff
@@ -0,0 +1,294 @@
+                    self.model = linevals[1]
+                    self.modeldate = linevals[2]
+                elif len(linevals) == 6:
+                    degree_n = int(float(linevals[0]))
+                    order_m = int(float(linevals[1]))
+                    gauss_g = float(linevals[2])
+                    gauss_h = float(linevals[3])
+                    gauss_g_dot = float(linevals[4])
+                    gauss_h_dot = float(linevals[5])
+                    self.coefficient[order_m][degree_n] = gauss_g
+                    self.coefficient_dot[order_m][degree_n] = gauss_g_dot
+                    if (order_m != 0):
+                        self.coefficient[degree_n][order_m - 1] = gauss_h
+                        self.coefficient_dot[degree_n][order_m - 1] = gauss_h_dot
+        self._unnormalise_gauss_coefficients()
+        
+    def _unnormalise_gauss_coefficients(self):
+        ''' Convert Schmidt normalized Gauss coefficients to unnormalized '''
+        schmidt_norm = schmidt_quasi_normilisation(self.array_size)
+        for n in range(self.array_size):
+            for m in range(self.array_size):
+                if m <= n:
+                    self.coefficient[m][n] = schmidt_norm[m][n] * self.coefficient[m][n]
+                    self.coefficient_dot[m][n] = schmidt_norm[m][n] * self.coefficient_dot[m][n]
+                else:
+                    self.coefficient[m][n] = schmidt_norm[n + 1][m] * self.coefficient[m][n]
+                    self.coefficient_dot[m][n] = schmidt_norm[n + 1][m] * self.coefficient_dot[m][n]
+
+    def _time_adjust_gauss(self, time):
+        '''Time adjust the Gauss Coefficients
+        
+            There is a very basic cache happening here where if the time delta 
+            hasn't changed then the previous calculation is returned
+        '''
+        current_delta_time = _calculate_decimal_year(time) - self.epoch
+        if self._last_calculated_datetime != current_delta_time:
+            self._update_time_coefficients(current_delta_time)
+            self._last_calculated_datetime = current_delta_time
+        return self.time_adjusted_coefficients_cache
+
+    def _update_time_coefficients(self, delta_time):
+        for n in range(1, self.max_order + 1):
+            for m in range(0, n + 1):
+                self.time_adjusted_coefficients_cache[m][n] = (self.coefficient[m][n] +
+                                                         delta_time * self.coefficient_dot[m][n])
+                if (m != 0):
+                    self.time_adjusted_coefficients_cache[n][m - 1] = (self.coefficient[n][m - 1] +
+                                                                 delta_time * self.coefficient_dot[n][m - 1])
+
+
+class WorldMagneticModel:
+    '''Class for calculating geomagnetic variation according to the world magnetic model
+
+    Example Usage:
+
+    >>> from geomag import WorldMagneticModel
+    >>> WorldMagneticModel().calc_mag_field(80,0).declination
```

Comment:
Ah, you're referring to the output. I assumed it was the method call based on the lines shown above. I haven't run doctest on it, I guess removing it is the way to clear it up.